src/createRollupConfig.ts

```diff
@@ -125,22 +126,16 @@ export function createRollupConfig(opts: TsdxOptions) {
           };
         },
       },
-      typescript({
-        typescript: require('typescript'),
-        cacheRoot: `./.rts2_cache_${opts.format}`,
-        tsconfig: opts.tsconfig,
-        tsconfigDefaults: {
-          compilerOptions: {
-            sourceMap: true,
-            declaration: true,
-            jsx: 'react',
-          },
-        },
-        tsconfigOverride: {
-          compilerOptions: {
-            target: 'esnext',
-          },
-        },
+      ts({
+        tsconfig: tsconfig => ({
+          ...tsconfig,
+          target: ScriptTarget.ESNext,
+          sourceMap: true,
+          declaration: true,
+          jsx: JsxEmit.React,
+        }),
+        // transpiler: 'babel',      // nothing seems to make last two tests pass.
```

Comment:
Uncomment it, and also pass the desired default babel config as `babelConfig`.

But I'm not sure if you pass babelConfig option does it will support project's babel configurations. This `ts` plugin is doing A TON of magic behind the scenes. It generally will work without config, and even if the project's babel.config.js has some presets it will override some of their options. 

A lot of magic is happening which I don't like. Just look how big the codebase is.

Reply:
Hey there, I'm the author of the package. I'm glad to assist you with any questions you may have during migration. Yes, it's true that some Babel options are forced to ensure maximum interoperability with Rollup. I've documented all of these forced options [here](https://github.com/wessberg/rollup-plugin-ts/blob/master/README.md#ignoredoverridden-babel-options) and [here](https://github.com/wessberg/rollup-plugin-ts/blob/master/README.md#combining-typescript-with-babel).

If there are any forced options that you find questionable, please let me know so I can take a look at it.